### PR TITLE
Add environment banner to non-production apps

### PR DESCRIPTION
**Pull request by csutter**
This will help us distinguish between our myriad non-production apps at
a glance, and make it clear to internal users that they are not on
production for peace of mind when performing destructive actions.

- Add `APP_ROLE` environment variable to distinguish what we call an
  "environment" for operational purposes from Rails environment (which
  is always set to `production` outside of local development)
- Add `Rails.configuration.app_role` to allow for easy checking of
  current app role
- Set `APP_ROLE` for all existing environments
- Create a spiffy environment banner that is displayed when the app
  role is a) set and b) not "production"
- Include environment banner in global application layout

Screenshot 2021-03-17 at 11 37 49

### CODE REVIEW

File: app/components/shared/environment_banner_component.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Shared::EnvironmentBannerComponent < ViewComponent::Base
```
Comment (cpjmcquillan) [inline app/components/shared/environment_banner_component.rb:1]: How do you feel about extending `GovukComponent::Base`?

**Review by cpjmcquillan — APPROVED** [review]
💥  do you want to add a spec?

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3093.london.cloudapps.digital>

File: app/components/shared/environment_banner_component.rb (hunk 1 above)
Comment (csutter) [reply app/components/shared/environment_banner_component.rb:1]: Sure! This was just copypasta'ed from another component 😅

File: spec/components/shared/environment_banner_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    context "when app_role is production" do
+      let(:app_role) { "production" }
+
+      it "does not render" do
+        expect(subject.render?).to eq(false)
+      end
+    end
+
+    context "when app_role is unknown" do
```
Comment (cpjmcquillan) [inline spec/components/shared/environment_banner_component_spec.rb:17]: if we do a `render_inline` somewhere then we can do things like `expect(subject).to be_rendered` vice versa, but I'm happy enough if you just want to test the method itself instead

**Pull request merged**